programs/ziti-edge-tunnel/CMakeLists.txt

```diff
@@ -33,6 +33,19 @@ endif ()
 add_executable(ziti-edge-tunnel ziti-edge-tunnel.c ${NETIF_DRIVER_SOURCE} ${ZITI_INSTANCE_COMMON} ${ZITI_INSTANCE_OS})
 set_property(TARGET ziti-edge-tunnel PROPERTY C_STANDARD 11)
 
+option(LIBSYSTEMD_FEATURE_FLAG "libsystemd library integration toggle" ON)
+
+if(CMAKE_SYSTEM_NAME STREQUAL "Linux" AND LIBSYSTEMD_FEATURE_FLAG)
+    find_package(PkgConfig REQUIRED)
+
+    pkg_check_modules(LIBSYSTEMD IMPORTED_TARGET "libsystemd")
+
+    if(LIBSYSTEMD_FOUND AND LIBSYSTEMD_VERSION VERSION_GREATER_EQUAL "229")
```

Comment:
Is this comment resolved? I think all of the libsystemd checks/additions need to be removed from this PR. Otherwise the resulting executable will have a hard requirement on libsystemd.so, and I think that's what we want to avoid?

Also in general it's probably not a good idea to assume anything about the system that the binary is going to be executed on based on what's available on the build system. They are usually entirely different systems (except sometimes in the case of the development host)

Reply:
And I think it's acceptable for the build to fail if the build host doesn't have the libsystemd includes (or the right version of them)

Reply:
I believe it is resolved, yes. 
In this case, the builder system need not have libsystemd to compile this code. This check now only sets `USE_LIBSYSTEMD_RESOLVER`, and only when an appropriate version is found on the builder system, affording us  a path compile on older distributions (like RHEL7, and Ubuntu Trusty). Preprocessor #ifdef's conditionally include the required sections of code appropriate. That is, we can build conditionally. 

Then, we also run conditionally. If a binary is built on a system that does contain libsystemd, the symbols are loaded dynamically at runtime. If we cannot dlopen libsystemd.so or dlsym one of the needed functions, we fall back to a different resolver.

Reply:
> This check now only sets USE_LIBSYSTEMD_RESOLVER, and only when an appropriate version is found on the builder system

This is what I think we want to avoid. The supported features of the executable are determined by the configuration of the build system. In my opinion, it's preferable to know what will be in a build without having intimate knowledge of the build system.

I think the cmakefiles should produce a fully capable executable by default, unless perhaps a flag was specified to deliberately exclude a feature. I also think it's fair to expect the systemd includes to be present on the build host, and let the build fail if they are absent. I'm assuming the build will also fail if the version of the systemd includes is insufficient. If that is not the case then we could force a build failure when the version isn't what we want.